screens/DoubleSearch.js

```diff
@@ -42,27 +42,54 @@ function DoubleSearch(props) {
     const [coordinatesFrom, setCoordinatesFrom] = React.useState(null);
     const [coordinatesTo, setCoordinatesTo] = React.useState("");
     const [currentLocationCoords, setCurrentLocationCoords] = React.useState(null);
+    const [pointOfInterest, setPointOfInterest] = React.useState(null);
+    const [coordinatesPOI, setCoordinatesPOI] = React.useState({ latitude: null, longitude: null });
+
+
+    const namePointOfInterest = props.navigation.getParam("name_POI", null);
+    const latitudePointOfInterest = props.navigation.getParam("latitude_POI", null);
+    const longitudePointOfInterest = props.navigation.getParam("longitude_POI", null);
+    
+    /**
+     * Adding the point of interest as an item in the dropdownlist of the Destination search bar
+     */
+    const addItem = () => {
+            destinationItems.unshift({ id: 0, name: namePointOfInterest});   
+    }
+    const [courseScheduleLocation, setCourseScheduleLocation] = React.useState("");
+    const [initialTo, setInitialTo] = React.useState("");
+    const [placeholder, setPlaceholder] = React.useState("");
+
+
     /**
-     * Description: Method to go back to the previous screen.
+     * Description: Method to back to the previous screen.
      * Using Stack navigator.
      */
 
     // var fromScreen; 
-    const CourseScheduleDetailsScreen = props.navigation.getParam("CourseScheduleDetailsScreen", "null");
+    const CourseSchedule = props.navigation.getParam("CourseScheduleScreen", "null");
+    const CourseScheduleLocation = props.navigation.getParam("CourseScheduleLocation", "");
     const NearbyInterestDetailsScreen = props.navigation.getParam("NearbyInterestDetailsScreen", "null");
+    const InitialTo = props.navigation.getParam("destinationName", "");
+
+
     const goBack = () => {
-        if(CourseScheduleDetailsScreen === true){
+        if (CourseSchedule === true) {
             props.navigation.goBack();
```

Comment:
why Having _goBack_() then call _navigate_?

Reply:
Its a work around to pop out the stack for _routeStack_. 

Since we're coming in from another stack, if I don't pop it, by navigating back to the _Map_ tab in the drawer, it will still render the DoubleSearch screen instead of the Map Screen

Reply:
Fixed!